hy/completer.py

```diff
@@ -83,29 +87,37 @@ def attr_matches(self, text):
         except Exception:
             return []
 
-        # Build match list to return
         n = len(attr)
         matches = []
         for w in words:
             if w[:n] == attr:
-                matches.append("%s.%s" % (expr, w.replace("_", "-")))
+                matches.append("{}.{}".format(
+                    expr.replace("_", "-"), w.replace("_", "-")))
         return matches
 
     def global_matches(self, text):
-        path = PATH
-        if self.namespace:
-            path.append(self.namespace)
+        matches = []
+        for p in self.path:
+            for k in p.keys():
+                k = str(k).replace("_", "-")
```

Comment:
str -> six.text_type. Did you test it with non-ASCII names? https://github.com/algernon/adderall/ has a couple of examples like `flattenrevᵒ`.